.circleci/config.yml

```diff
@@ -1,171 +1,60 @@
-          root: ~/
-          paths:
-            - workdir
-
-  build:
-    machine: true
-    environment:
-      <<: *host_environment_defaults
-    working_directory: ~/workdir
-    steps:
-      - attach_workspace:
-          at: ~/
-      - *setup_dependencies_key
-      - *restore_dependencies
-      - run:
-          name: Clone test project
-          working_directory: ~/
-          command: |
-            (git -C ~/codacy-plugins-test fetch --all &&
-              git -C ~/codacy-plugins-test reset --hard origin/master) ||
-                git clone git://github.com/codacy/codacy-plugins-test.git ~/codacy-plugins-test
-      - run:
-          name: Compile test project
-          working_directory: ~/codacy-plugins-test
-          command: sbt compile
-      - run:
-          name: Publish tool docker locally
-          working_directory: ~/workdir
-          command: ./scripts/publish.sh $CIRCLE_PROJECT_REPONAME latest
-      - save_cache:
-          key: sbt-ivy2-dependencies-1.0.0-{{ checksum "/tmp/dependencies.cache.tmp" }}
-          paths:
-            - "~/.ivy2"
-            - "~/.m2"
-            - "~/.sbt"
-            - "~/codacy-plugins-test/target"
-            - "~/codacy-plugins-test/project/target"
-            - "~/codacy-plugins-test/project/project"
-            - "~/workdir/target"
-            - "~/workdir/project/target"
-            - "~/workdir/project/project"
-      - run:
-          name: Test json
-          working_directory: ~/codacy-plugins-test
-          command: sbt -Dcodacy.tests.ignore.descriptions=true "runMain codacy.plugins.DockerTest json $CIRCLE_PROJECT_REPONAME:latest"
-      - run:
-          name: Test patterns
-          working_directory: ~/codacy-plugins-test
-          command: sbt -Dcodacy.tests.noremove=true -Dcodacy.tests.threads=8 "runMain codacy.plugins.DockerTest pattern $CIRCLE_PROJECT_REPONAME:latest"
-      - deploy:
-          name: Push application Docker image
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              docker login -u $DOCKER_USER -p $DOCKER_PASS
-              docker tag $CIRCLE_PROJECT_REPONAME codacy/$CIRCLE_PROJECT_REPONAME
-              docker tag $CIRCLE_PROJECT_REPONAME codacy/$CIRCLE_PROJECT_REPONAME:$(cat .version)
-              docker push codacy/$CIRCLE_PROJECT_REPONAME
-              docker push codacy/$CIRCLE_PROJECT_REPONAME:$(cat .version)
-            fi
-
+  codacy: codacy/base@1.0.0
+  codacy_plugins_test: codacy/plugins-test@0.4.0
 workflows:
   version: 2
-  build-and-deploy:
+  compile_test_deploy:
     jobs:
-      - checkout_and_version
-      - sbt_dependencies:
+      - codacy/checkout_and_version:
+          write_sbt_version: true
+      - codacy/sbt:
+          name: populate_cache
+          cmd: |
+            sbt "set scalafmtUseIvy in ThisBuild := false;
+                 set name := \"$CIRCLE_PROJECT_REPONAME\";
+                 update"
+          persist_to_workspace: true
```

Comment:
You can shrink this step in the other as we are doing for all the other tools

Reply:
Done